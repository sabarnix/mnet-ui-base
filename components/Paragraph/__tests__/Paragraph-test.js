"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

require("jest-styled-components");

var _MnetUIBase = require("../../MnetUIBase");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

test('Paragraph renders', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Paragraph, null)));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Paragraph size renders', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    size: "small"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    size: "medium"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    size: "large"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    size: "xlarge"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    size: "xxlarge"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    fill: true
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    fill: false
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Paragraph margin renders', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: "small"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: "medium"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: "large"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: "none"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: {
      bottom: 'small'
    }
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    margin: {
      top: 'small'
    }
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Paragraph textAlign renders', function () {
  var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    textAlign: "start"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    textAlign: "center"
  }), /*#__PURE__*/_react["default"].createElement(_.Paragraph, {
    textAlign: "end"
  })));

  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});