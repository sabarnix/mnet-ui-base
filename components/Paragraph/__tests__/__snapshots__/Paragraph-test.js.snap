// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph margin renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 4px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c2 {
  margin: 8px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c3 {
  margin: 16px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c4 {
  margin: 0px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c5 {
  margin-bottom: 4px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c6 {
  margin-top: 4px;
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

<div
  className="c0"
>
  <p
    className="c1"
  />
  <p
    className="c2"
  />
  <p
    className="c3"
  />
  <p
    className="c4"
  />
  <p
    className="c5"
  />
  <p
    className="c6"
  />
</div>
`;

exports[`Paragraph renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

<div
  className="c0"
>
  <p
    className="c1"
  />
</div>
`;

exports[`Paragraph size renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 9.333333333333334px;
  line-height: 13.333333333333334px;
  max-width: 149.33333333333334px;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
}

.c3 {
  font-size: 14.666666666666666px;
  line-height: 18.666666666666668px;
  max-width: 234.66666666666666px;
}

.c4 {
  font-size: 17.333333333333332px;
  line-height: 21.333333333333332px;
  max-width: 277.3333333333333px;
}

.c5 {
  font-size: 22.666666666666664px;
  line-height: 26.666666666666664px;
  max-width: 362.66666666666663px;
}

.c6 {
  font-size: 12px;
  line-height: 16px;
  max-width: none;
}

<div
  className="c0"
>
  <p
    className="c1"
    size="small"
  />
  <p
    className="c2"
    size="medium"
  />
  <p
    className="c3"
    size="large"
  />
  <p
    className="c4"
    size="xlarge"
  />
  <p
    className="c5"
    size="xxlarge"
  />
  <p
    className="c6"
  />
  <p
    className="c2"
  />
</div>
`;

exports[`Paragraph textAlign renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  text-align: left;
}

.c2 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  text-align: center;
}

.c3 {
  font-size: 12px;
  line-height: 16px;
  max-width: 192px;
  text-align: right;
}

<div
  className="c0"
>
  <p
    className="c1"
  />
  <p
    className="c2"
  />
  <p
    className="c3"
  />
</div>
`;
